html/embed/index.js

```diff
@@ -139,4 +159,76 @@ function handleTsParam(ts) {
   }
 
   return ts;
-}
\ No newline at end of file
+}
+
+/**
+ * Waits for a given element to be in the DOM and returns it.
+ * Wait is based on `requestAnimationFrame`: timeout is approximately 60 seconds (60 x 60 frames per seconds).
+ *
+ * Takes a function querying the DOM for a single element as an argument
+ */
+async function waitForElement(selectorFunction) {
+  const maxPauseSeconds = 60
+  let tries = maxPauseSeconds * 60;  // we expect a repaint rate of ~60 times a second, per https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame
+  let elem = null;
+
+  while (!elem && tries > 0) {
+    // Sleep efficiently until the next repaint
+    const pause = await new Promise(resolve => requestAnimationFrame(resolve));
+    cancelAnimationFrame(pause);
+
+    // Look for the target element
+    try {
+      elem = selectorFunction();
+    }
+    catch (err) {
+      if (!err.message.includes('null')) {
+        throw err;
+      }
+      tries -= 1;
+    }
+  }
+
+  if (elem) {
+    return elem;
+  }
+
+  throw new Error("Timed out");
```

Comment:
About the suggestion we were discussing to avoid having to rely on exception messages: https://javascript.info/custom-errors#extending-error 

I would maybe create an exception type, but check for the presence / absence of a property.

Reply:
Gah! Clearly I have not retained everything I learned when reading javascript.info 😄 . Thank you yes, I will fix this.